### PR TITLE
Finish love.audio

### PR DESCRIPTION
**Pull request by Positive07**
Made sure that love.audio is up-to-date with LÖVE's source code.

There is stuff that isn't even documented in the wiki which I'll list down later in the comments.

Type checking and proof reading would be appreciated! :heart:

### CODE REVIEW

**Comment by Positive07** [comment]
I'm hurrying this PR cause I saw you push `SourceType` @rm-code and I don't want you to do extra work (I had those changes included, but I re-based and used yours)

**Comment by rm-code** [comment]
Looks good even though I only did a quick check. I'd like to wait to pull until the wiki is updated aswell.

**Comment by rm-code** [comment]
If you have the time, a list with the stuff that needs to be updated on the wiki would help a lot 👍

File: modules/audio/types/Source.lua [hunk 1]
```diff
@@ -75,6 +75,11 @@ return {
                             type = 'number',
                             name = 'outerVolume',
                             description = 'The Source\'s volume when the listener is outside both the inner and outer cone angles.'
+                        },
+                        {
+                            type = 'number',
+                            name = 'outerHighGain',
```
[the remark this answers is not in the document]
Comment (Positive07) [reply modules/audio/types/Source.lua]: Note this parameter will be 0 always if the system doesn't support a high-gain change.

File: modules/audio/types/Source.lua [hunk 2]
```diff
@@ -397,6 +539,153 @@ return {
                             name = 'outerVolume',
                             default = '0',
                             description = 'The Source\'s volume when the listener is outside both the inner and outer cone angles.'
+                        },
+                        {
+                            type = 'number',
+                            name = 'outerHighGain',
```
[the remark this answers is not in the document]
Comment (Positive07) [reply modules/audio/types/Source.lua]: Note this parameter will be ignored if the system doesn't support changing the high-gain.

File: modules/audio/Audio.lua [hunk 3]
```diff
@@ -92,15 +88,30 @@ return {
...
-                            description = 'The number of sources which are currently playing or paused.'
+                            name = 'maximum',
+                            description = 'The maximum number of active Effects.'
+                        }
+                    }
+                }
+            }
+        },
+        {
+            name = 'getMaxSourceEffects',
```
[the remark this answers is not in the document]
Comment (MikuAuahDark) [reply modules/audio/Audio.lua]: Sorry for interruption but this one doesn't actually exist in `love.audio`. Can you elaborate?

File: modules/audio/Audio.lua [hunk 4]
```diff
@@ -92,15 +88,30 @@ return {
             }
         },
         {
-            name = 'getSourceCount',
-            description = 'Returns the number of sources which are currently playing or paused.',
+            name = 'getMaxActiveEffects',
```
[the remark this answers is not in the document]
Comment (Positive07) [reply modules/audio/Audio.lua]: This is called getMaxSceneEffects, I'll PR a fix

**Pull request merged**